src/shared/nested-dropdown-list/nested-dropdown-list.tsx

```diff
@@ -0,0 +1,326 @@
+            setContentHeight(getContentHeight());
+        });
+    };
+
+    function handleKeyboardPress(event: KeyboardEvent) {
+        if (nodeRef && (nodeRef.current as any).contains(event.target)) {
+            switch (event.code) {
+                case "ArrowDown":
+                    break;
+                case "ArrowUp":
+                    break;
+                case "Escape":
+                    if (onDismiss) onDismiss(true);
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+
+    const handleBlur = () => {
+        if (onBlur) {
+            handleBlur;
+        }
+    };
+
+    const handleTryAgain = () => {
+        if (onRetry) {
+            onRetry();
+        }
+    };
+
+    // =============================================================================
+    // HELPER FUNCTIONS
+    // =============================================================================
+    const getContentHeight = () => {
+        const listHeight =
+            listRef && listRef.current
+                ? listRef.current.getBoundingClientRect().height
+                : 0;
+
+        return listHeight;
+    };
+
+    const getDefaultExpandKeys = (
+        items: Map<string, FItemOption<V1, V2, V3>> | undefined,
+        parentKeys?: string[]
+    ): string[] => {
+        /**
+         * Get the keys which has the subItems on "items" tier
+         */
+        let keys = [];
+
+        if (!items) return keys;
+
+        if (parentKeys && parentKeys.length) {
+            keys = parentKeys;
+        }
+
+        for (const [key, item] of items.entries()) {
+            if (item.subItems && item.subItems.size) {
+                keys.push(key);
+
+                getDefaultExpandKeys(item.subItems, keys);
+
+                // stop loop once hit the innermost level
+                break;
+            }
+        }
+
+        return keys;
+    };
+
+    const getItemWithSubItems = (
+        items: Map<string, FItemOption<V1, V2, V3>>
+    ) => {
+        /**
+         * Filter the category which has the subItems
+         */
+        const itemWithSubItems = [];
+
+        for (const [key, value] of items.entries()) {
+            if (value.subItems && value.subItems.size) {
+                const item = new Map();
+
+                item.set(key, value);
+                itemWithSubItems.push(item);
+
+                // stop loop once found the item
+                break;
+            }
+        }
+
+        return itemWithSubItems;
+    };
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    const renderItems = () => {
+        if (!onRetry || (onRetry && itemsLoadState === "success")) {
+            const [item] = getItemWithSubItems(listItems);
+            const defaultExpandKeys = getDefaultExpandKeys(item);
```

Comment:
this doesn't seem ideal for cases where the items have to be collapsed/expanded programmatically (e.g. for the search functionality)

we previously discussed maintaining a map of the items' expanded state, is that something you are considering in subsequent PRs?

Reply:
okay, understand that expand/collapse handling will be covered in the next PR, so I'll take the current implementation